rudder-core/src/main/scala/com/normation/rudder/domain/queries/CmdbQuery.scala

```diff
@@ -406,32 +406,38 @@ case object OsNameComparator extends CriterionType {
 case object AgentComparator extends CriterionType {
 
   val ANY_CFENGINE = "cfengine"
-  /*
-   * We are using 'oldShortName' because we are matching on substring,
-   * and that name was used in the past.
-   */
-  val agentTypes = ( (ANY_CFENGINE, "Any CFEngine based agent") :: AgentType.allValues.toList.map(a => (a.oldShortName, a.displayName))).sortBy( _._2 )
+  val (cfeTypes, cfeAgents) = ((ANY_CFENGINE, "Any CFEngine based agent"),(ANY_CFENGINE, AgentType.CfeCommunity :: AgentType.CfeEnterprise :: Nil))
+  val allAgents = AgentType.allValues.toList
+
+  val agentTypes = ( cfeTypes  :: allAgents.map(a => (a.id, (a.displayName)))).sortBy( _._2 )
+  val agentMap   = ( cfeAgents :: allAgents.map(a => (a.id, a :: Nil))).toMap
 
   override def comparators = Seq(Equals, NotEquals)
   override protected def validateSubCase(v:String,comparator:CriterionComparator) = {
     if(null == v || v.length == 0) Failure("Empty string not allowed") else Full(v)
   }
   override def toLDAP(value:String) = Full(value)
 
-  private[this] def filterAgent(agent: String) = SUB(A_AGENTS_NAME, null, Array(agent), null)
+  private[this] def filterAgent(agent: AgentType) =
+      SUB(A_AGENTS_NAME, null, Array(s""""agentType":"${agent.id}""""), null) ::
+      EQ(A_AGENTS_NAME, agent.oldShortName) ::
```

Comment:
Could you add a comment here so that we remember why we need to keep old short name here?